### PR TITLE
chore: do not print unrelated logs

### PR DESCRIPTION
**Pull request by KazuCocoa**
These logs are not used by these two caps, but could confuse users

### CODE REVIEW

**Review by mykola-mokhnach — APPROVED** [review]

File: lib/driver.js [hunk 1]
```diff
@@ -770,11 +770,13 @@ class XCUITestDriver extends BaseDriver {
         await this.wda.setupCaching();
       }
 
+      const noXcodebuildCaps = ['webDriverAgentUrl', 'usePreinstalledWDA']
```
Comment (mykola-mokhnach) [inline lib/driver.js:773]: should we also assign `shortCircuitError` in such case and exit from the wdaStartupRetries loop to save time?

Comment (mykola-mokhnach) [reply lib/driver.js:773]: I think we could just use `isXcodebuildNeeded` here and then always shortcircuit if that's the case

Comment (KazuCocoa) [reply lib/driver.js]: makes sense

File: lib/driver.js [hunk 2]
```diff
@@ -762,8 +762,7 @@ class XCUITestDriver extends BaseDriver {
       // local helper for the two places we need to uninstall wda and re-start it
       const quitAndUninstall = async (msg) => {
         this.log.debug(msg);
-        const noUninstallCap = ['webDriverAgentUrl', 'usePreinstalledWDA']
-          .find((x) => Boolean(this.opts[x]));
+        const noUninstallCap = !this.isXcodebuildNeeded();
```
Comment (mykola-mokhnach) [inline lib/driver.js:765]: I would say we don't need this intermediate variable here. Simply move `!this.isXcodebuildNeeded()` directly to the if condition

File: lib/driver.js [hunk 3]
```diff
@@ -781,20 +780,27 @@ class XCUITestDriver extends BaseDriver {
...
+          !util.hasValue(this.opts.wdaStartupRetries) &&
+          !util.hasValue(this.opts.wdaStartupRetryInterval)
+        ) {
+          this.log.debug(
+            `These values can be customized by changing wdaStartupRetries/wdaStartupRetryInterval capabilities`,
+          );
+        }
+      } else {
+        // The startup retry will be one time if the session does not need WDA build
+        startupRetries = 1;
```
Comment (mykola-mokhnach) [inline lib/driver.js:803]: we probably should mention in docs for these caps that they only make sense if xcodebuild is used to compile/start WDA

File: lib/driver.js [hunk 4]
```diff
@@ -762,11 +762,9 @@ class XCUITestDriver extends BaseDriver {
       // local helper for the two places we need to uninstall wda and re-start it
       const quitAndUninstall = async (msg) => {
         this.log.debug(msg);
-        const noUninstallCap = ['webDriverAgentUrl', 'usePreinstalledWDA']
-          .find((x) => Boolean(this.opts[x]));
-        if (noUninstallCap) {
+        if (!this.isXcodebuildNeeded()) {
           this.log.debug(
-            `Not quitting/uninstalling WebDriverAgent since ${noUninstallCap} capability is provided`,
+            `Not quitting/uninstalling WebDriverAgent since ${!this.isXcodebuildNeeded()} capability is provided`,
```
Comment (mykola-mokhnach) [inline lib/driver.js:767]: here we actually need to put capability names :)
maybe move them into a constant, which could be shared between this message and the isXcodebuildNeeded helper

Comment (KazuCocoa) [reply lib/driver.js:767]: Wow, yea lol

File: lib/driver.js [hunk 5]
```diff
@@ -170,6 +170,9 @@ const NO_PROXY_WEB_LIST = /** @type {import('@appium/types').RouteMatcher[]} */
 
 const MEMOIZED_FUNCTIONS = ['getStatusBarHeight', 'getDevicePixelRatio', 'getScreenInfo'];
 
+// Capabilities that do not have xcodebuild process
+const CAP_NAMES_NO_XCODEBUILD_REQUIRED = ['webDriverAgentUrl', 'usePreinstalledWDA'];
```
Comment (mykola-mokhnach) [inline lib/driver.js:174]: Please also update isXcodebuildNeeded to use this constant

Comment (KazuCocoa) [reply lib/driver.js:174]: ba98cc8a972904e7fe8fc38fc8f60672b68c908e

File: lib/driver.js [hunk 6]
```diff
@@ -762,11 +765,9 @@ class XCUITestDriver extends BaseDriver {
       // local helper for the two places we need to uninstall wda and re-start it
       const quitAndUninstall = async (msg) => {
         this.log.debug(msg);
-        const noUninstallCap = ['webDriverAgentUrl', 'usePreinstalledWDA']
-          .find((x) => Boolean(this.opts[x]));
-        if (noUninstallCap) {
+        if (!this.isXcodebuildNeeded()) {
           this.log.debug(
-            `Not quitting/uninstalling WebDriverAgent since ${noUninstallCap} capability is provided`,
+            `Not quitting/uninstalling WebDriverAgent since at least one of ${CAP_NAMES_NO_XCODEBUILD_REQUIRED} capability is provided`,
```
Comment (mykola-mokhnach) [inline lib/driver.js:770]: one of ... capabilities

File: lib/driver.js [hunk 7]
```diff
@@ -382,7 +385,7 @@ class XCUITestDriver extends BaseDriver {
   }
 
   isXcodebuildNeeded() {
-    return !(this.opts.webDriverAgentUrl || this.opts.usePreinstalledWDA);
+    return !(CAP_NAMES_NO_XCODEBUILD_REQUIRED.find((x) => Boolean(this.opts[x])));
```
Comment (mykola-mokhnach) [inline lib/driver.js:388]: find could be replaces with some

File: lib/driver.js (hunk 6 above)
Comment (mykola-mokhnach) [reply lib/driver.js:770]: same below

**Review by mykola-mokhnach — APPROVED** [review]

**Pull request merged**